dip-0027.md

```diff
@@ -0,0 +1,163 @@
+
+## Overview
+
+When Dash is locked in the credit pool, the person who locked it can claim a corresponding number of credits for an Identity on Dash Platform. As the Identity performs actions on Platform, its balance is reduced by the cost of the actions while the masternode payout pool value is increased by the same amount. Over time, funds in the masternode payout pool are distributed to the Identities of masternodes that participated in Platform validator sets. The details of masternode Identities and Platform payouts will be the focus of future DIPs.
+
+Eventually, an Identity may want to withdraw their credits from Platform to regain access to the underlying locked Dash. Any Identity type can issue a withdrawal request to Dash Platform. If the Identity’s balance is sufficient, this will result in the formation of a transaction signed by a quorum and valid on the Core chain. This transaction unlocks the requested amount of Dash from the credit pool and sends it to the provided Dash address.
+
+## Asset Locking
+
+Dash Core participates in the credit creation process by locking Dash in the credit pool with an Asset Lock transaction.
+
+### Asset Lock Transaction
+
+The Asset Lock transaction (AssetLockTx) is a DIP-2-based special transaction that gives Platform the information required to assign credits to Identities. This special transaction is type 8.
+
+The format of the Asset Lock special transaction payload:
+
+| **Name** | **Type** | **Size**  | **Description** |
+| - | - | - | - |
+| version | uint8_t | 1 | Asset Lock transaction version number |
+| count | uint8_t | 1 | The number of accounts funded via this Asset Lock |
+| credit_outputs | vec\<TxOut> | 28 * count| The hash of the public key(s) used to claim credits on Dash Platform, as well as the amount of Dash in duffs transferred into credits for this public key hash |
+
+Each Asset Lock transaction must create an unspendable output by using an OP_RETURN as the first script. The following criteria apply to the OP_RETURN output:
+
+* It must not include any data
+* Its value is the amount of Dash to be locked
+* The sum of all TxOut amounts must equal the value of the output
+* It must be the only output in the transaction using an OP_RETURN
+
+### Proof of Finality
+
+It is important to note that Dash Core transactions achieve finality through ChainLocks and InstantSend locks. Therefore, these locks can be used to prove that value has been added to the credit pool and is ready for Platform use. For example, Platform can safely increase an Identity’s balance without concerns that the related Asset Lock transaction will be double-spent.
+
+There are two ways to prove transaction finality: either by including the InstantSend lock of a recent transaction or by referencing the chain height at which a transaction was included in a ChainLocked block. Typically, it is preferable to use the InstantSend lock for proof, but a ChainLock can be used if an InstantSend lock is not available.
```

Comment:
Actually, you can provide any height that is higher or equal to the block where the transaction was mined.

Reply:
See suggestion proposed below...